app/src/main/java/com/jerboa/ui/components/home/sidebar/SiteSidebarActivity.kt

```diff
@@ -14,24 +21,43 @@ import com.jerboa.ui.components.common.SimpleTopAppBar
 @OptIn(ExperimentalMaterial3Api::class)
 @Composable
 fun SiteSidebarActivity(
+    appState: JerboaAppState,
     siteViewModel: SiteViewModel,
-    onBackClick: () -> Unit,
 ) {
     Log.d("jerboa", "got to site sidebar activity")
 
     val title =
         when (val siteRes = siteViewModel.siteRes) {
-            is ApiState.Success -> "${siteRes.data.site_view.site.name} Info"
+            is ApiState.Success -> stringResource(R.string.site_info_name, siteRes.data.site_view.site.name)
             else -> {
-                "Loading..."
+                stringResource(R.string.loading)
             }
         }
 
     Scaffold(
         topBar = {
             SimpleTopAppBar(
                 text = title,
-                onBackClick,
+                onClickBack = appState::popBackStack,
+                actions = {
```

Comment:
Added it to the action bar of the site info page, rather than the top of home. Could go either way.